runtime/src/qio/qio_popen.c

```diff
@@ -337,9 +338,10 @@ qioerr qio_waitpid(int64_t pid,
 
   if( WIFEXITED(status) ) {
     *exitcode = WEXITSTATUS(status);
-    if( WIFSIGNALED(status) ) {
-      *exitcode = - WSTOPSIG(status);
-    }
+    *done = 1;
+  }
+  else if( WIFSIGNALED(status) ) {
+    *exitcode = WTERMSIG(status);
```

Comment:
This patch makes 2 changes here:
 1) Use WTERMSIG instead of WSTOPSIG
 2) sets exitcode to the signal number rather than negative signal number.
I think that 1 is great and fixes an oversight in the original implementation.
I think it would still be nice for the exit code to be a negative signal number, because normal exit codes can only be 0..255, so that would allow for someone calling wait to know whether a process exited with a signal.